examples/ubo_tilemap/main.rs

```diff
@@ -0,0 +1,525 @@
+                let vertex_y = half_height as f32 * raw_y;
+
+                let u_pos = (1.0 + raw_x) / 2.0;
+                let v_pos = (1.0 + raw_y) / 2.0;
+                let tilemap_x = (u_pos * width as f32) as i32;
+                let tilemap_y = (v_pos * height as f32) as i32;
+
+                VertexData {
+                    pos: [vertex_x, vertex_y, 0.0],
+                    buf_pos: [tilemap_x as f32, tilemap_y as f32]
+                }
+            })
+            .collect();
+
+        let index_data: Vec<u32> = plane.indexed_polygon_iter()
+            .triangulate()
+            .vertices()
+            .map(|i| i as u32)
+            .collect();
+        let slice = index_data.to_slice(factory, gfx::PrimitiveType::TriangleList);
+
+        // set up texture
+        let mesh = factory.create_mesh(&vertex_data);
+        // shader
+        let program = {
+            let vs = gfx::ShaderSource {
+                glsl_150: Some(include_bytes!("tilemap_150.glslv")),
+                .. gfx::ShaderSource::empty()
+            };
+            let fs = gfx::ShaderSource {
+                glsl_150: Some(include_bytes!("tilemap_150.glslf")),
+                .. gfx::ShaderSource::empty()
+            };
+            factory.link_program_source(vs, fs).unwrap()
+        };
+
+        let tile_texture = load_texture(factory, tilesheet_bytes).unwrap();
+        let tilemap_buf = factory.create_buffer_dynamic::<TileMapData>(TILEMAP_BUF_LENGTH, gfx::BufferRole::Uniform);
+
+        let data = Uniforms {
+            model: Matrix4::identity().into_fixed(),
+            view: Matrix4::identity().into_fixed(),
+            proj: cgmath::perspective(cgmath::deg(60.0f32), 
+                                      stream.get_aspect_ratio(),
+                                      0.1, 4000.0
+                                      ).into_fixed(),
+            tilesheet: (tile_texture, None),
+            tilemap: tilemap_buf,
+            world_size: [width as f32, height as f32, tile_size as f32],
+            tilesheet_size: [tilesheet_width as f32, tilesheet_height as f32, tilesheet_total_width as f32, tilesheet_total_height as f32],
+            offsets: [0.0, 0.0],
+            _r: std::marker::PhantomData,
+        };
+
+        let mut charmap_data = Vec::new();
+        for _ in (0..total_size) {
+            charmap_data.push(TileMapData::new_empty());
+        }
+        let mut item: TileMapPlane<R> = TileMapPlane {
+            data: charmap_data,
+            batch: gfx::batch::Full::new(mesh, program, data).unwrap()
+        };
+        item.batch.slice = slice;
+        item.batch.state = gfx::DrawState::new().depth(gfx::state::Comparison::LessEqual, true);
+
+        return item;
+    }
+    
+    pub fn update_data<TFactory>(&mut self, factory: &mut TFactory) where TFactory: Factory<R> {
+        if factory.update_buffer(&self.batch.params.tilemap, &self.data, 0).is_ok() == false { panic!("Failed to update glsl buffer"); }
```

Comment:
You can just use an `unwrap` or `except` here if you just want to abort on error.